src/plugins/ui_core_views/custom_colors.ts

```diff
@@ -76,16 +77,23 @@ export class CustomColorsPlugin extends UIPlugin<CustomColorState> {
 
   constructor(config: UIPluginConfig) {
     super(config);
-    for (const color of config.customColors ?? []) {
-      this.tryToAddColor(color);
-    }
+    this.tryToAddColors(config.customColors ?? []);
   }
 
-  handle(cmd: CoreViewCommand) {
+  handle(cmd: Command) {
     switch (cmd.type) {
-      case "UPDATE_CELL":
+      case "START":
+        for (const sheetId of this.getters.getSheetIds()) {
+          for (const chartId of this.getters.getChartIds(sheetId)) {
+            this.tryToAddColors(this.getChartColors(chartId));
```

Comment:
shouldn't it concern every color source?

Reply:
the other ones are computed at every finalize, so we shouldn't need to bother with them there